pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -181,38 +190,58 @@ private Representation getSegmentFile(
   private Representation getSegmentsForTable(
       @Parameter(name = "tableName", in = "path", description = "The name of the table for which to list segments", required = true)
       String tableName,
-      @Parameter(name = "active", in = "query", description = "true = show active segments (in Helix), false = all segments (on filesystem)", required = false)
-      boolean activeOnly
-      ) {
+      @Parameter(name = "tableType", in = "query", description = "Type of table {offline|realtime}", required = false)
+      String type
+      ) throws Exception {
     Representation presentation;
+    JSONArray ret = new JSONArray();
+    final String realtime = "REALTIME";
+    final String offline = "OFFLINE";
+
+    if (type == null) {
+      ret.put(formatSegments(tableName, offline));
+      ret.put(formatSegments(tableName, realtime));
+    } else if (type.equalsIgnoreCase(realtime)) {
+      ret.put(formatSegments(tableName, realtime));
+    } else {
+      ret.put(formatSegments(tableName, offline));
+    }
+
+    presentation = new StringRepresentation(ret.toString());
+    return presentation;
+  }
+
+  private org.json.JSONObject formatSegments(String tableName, String tableType) throws Exception {
+    org.json.JSONObject obj = new org.json.JSONObject();
+    obj.put(tableType, getSegments(tableName, tableType));
+    return obj;
+  }
+  private JSONArray getSegments(String tableName, String tableType) {
+
     final JSONArray ret = new JSONArray();
 
-    if (activeOnly) {
-      String offlineTableName = TableNameBuilder.OFFLINE_TABLE_NAME_BUILDER.forTable(tableName);
-      List<String> segmentList = _pinotHelixResourceManager.getAllSegmentsForResource(offlineTableName);
-      ZkHelixPropertyStore<ZNRecord> propertyStore = _pinotHelixResourceManager.getPropertyStore();
+    String realtimeTableName = TableNameBuilder.REALTIME_TABLE_NAME_BUILDER.forTable(tableName);
```

Comment:
Please use TableType.valueOf(tableType) to derive the table type enum. Better yet, pass the enum to this method instead of string, so you can validate it in the calling method.